src/Icicle/Source/ToCore/ToCore.hs

```diff
@@ -132,19 +134,13 @@ convertQuery q
     -- We could support "older than" by storing reduce result of the end of window and
     -- storing all corresponding newer thans in the snapshot, so if this ends up being an issue
     -- we can address it.
-    (Windowed ann newerThan olderThan : _)
+    (Windowed _ newerThan olderThan : _)
      -> do  (bs, b) <- convertQuery q'
-            now <- convertDateName
-            (inp,_) <- convertInput
-
-            fs <- convertFeatures
-            let f = featureContextFactTime fs
-            fact <- case Map.lookup f (featureContextVariables fs) of
-                        Just f' -> return $ featureVariableExp f' $ CE.xVar inp
-                        Nothing -> convertError $ ConvertErrorExpNoSuchVariable (annAnnot ann) f
+            now  <- convertDateName
+            time <- convertFactTimeName
```

Comment:
this fixes #327: it was trying to get the second of the "input stream", but the local input stream in that case was a let-binding returning a pair ((fact,time),True) or whatever instead of the global, original input.
now it will just use the fact time.